File: caseflow-ai/backend/app/api/routes/evidence.py
from typing import List
from fastapi import APIRouter, Depends, UploadFile, File, Form, HTTPException
from sqlalchemy.orm import Session
from app.core.database import get_db
from app.services.evidence_service import EvidenceService
from app.schemas.evidence import EvidenceResponse
from app.schemas.evidence_extraction import EvidenceExtractionResponse

router = APIRouter(tags=["Evidence"])

@router.post("/cases/{id}/evidence", response_model=EvidenceResponse)
async def upload_case_evidence(
    id: str,
    file: UploadFile = File(...),
    evidence_type: str = Form("RECEIPT"),
    source_description: str = Form(None),
    db: Session = Depends(get_db)
):
    service = EvidenceService(db)
    file_bytes = await file.read()
    try:
        evidence = await service.upload_evidence(
            case_id=id,
            original_filename=file.filename or "uploaded_file",
            content_type=file.content_type or "application/octet-stream",
            file_bytes=file_bytes,
            evidence_type=evidence_type,
            source_description=source_description
        )
        return evidence
    except Exception as e:
        raise HTTPException(status_code=400, detail=str(e))

@router.get("/cases/{id}/evidence", response_model=List[EvidenceResponse])
def list_case_evidence(id: str, db: Session = Depends(get_db)):
    service = EvidenceService(db)
    return service.list_by_case(id)

@router.get("/evidence/{id}", response_model=EvidenceResponse)
def get_evidence(id: str, db: Session = Depends(get_db)):
    service = EvidenceService(db)
    evidence = service.get_by_id(id)
    if not evidence:
        raise HTTPException(status_code=404, detail="Evidence not found")
    return evidence

@router.post("/evidence/{id}/analyze", response_model=EvidenceExtractionResponse)
async def analyze_evidence(id: str, db: Session = Depends(get_db)):
    service = EvidenceService(db)
    try:
        extraction = await service.analyze_evidence(id)
        return extraction
    except Exception as e:
        raise HTTPException(status_code=400, detail=str(e))

File: caseflow-ai/frontend/src/api/evidence.ts
import { apiClient } from './client';
import { Evidence, EvidenceExtraction } from '../types';

export const uploadEvidence = async (
  caseId: string,
  file: File,
  evidenceType: string = 'RECEIPT',
  sourceDescription?: string
): Promise<Evidence> => {
  const formData = new FormData();
  formData.append('file', file);
  formData.append('evidence_type', evidenceType);
  if (sourceDescription) {
    formData.append('source_description', sourceDescription);
  }

  const response = await apiClient.post<Evidence>(`/cases/${caseId}/evidence`, formData, {
    headers: { 'Content-Type': 'multipart/form-data' },
  });
  return response.data;
};

export const getEvidenceList = async (caseId: string): Promise<Evidence[]> => {
  const response = await apiClient.get<Evidence[]>(`/cases/${caseId}/evidence`);
  return response.data;
};

export const analyzeEvidence = async (evidenceId: string): Promise<EvidenceExtraction> => {
  const response = await apiClient.post<EvidenceExtraction>(`/evidence/${evidenceId}/analyze`);
  return response.data;
};
